src/providers/extraction_providers/visionExtractionProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { ExtractionProvider } from './extractionProvider';
+import { Image, ExtractedImage } from '../../models/data_models/image';
+import { AssignedLabel, DetectedObject, Coordinate } from '../../models/data_models/extraction';
+
+import * as vision from '@google-cloud/vision';
+
+const enum requestTypes {
+	LabelDetection = 'LABEL_DETECTION',
+	ObjectLocalization = 'OBJECT_LOCALIZATION',
+}
+
+type LocalizedObjectAnnotation = vision.protos.google.cloud.vision.v1.ILocalizedObjectAnnotation;
+type EntityAnnotation = vision.protos.google.cloud.vision.v1.IEntityAnnotation;
+type NormalizedVertex = vision.protos.google.cloud.vision.v1.INormalizedVertex;
+
+export class VisionExtractionProvider implements ExtractionProvider {
+	protected createAssignedLabel(label: EntityAnnotation): AssignedLabel {
+		return {
+			description: label?.description || '',
+			confidence: label?.score || 0,
+		};
+	}
+
+	protected static createBoundingVertex(coordinate: NormalizedVertex): Coordinate {
+		return {
+			x: coordinate?.x || 0,
+			y: coordinate?.y || 0,
+		};
+	}
+
+	protected createDetectedObject(object: LocalizedObjectAnnotation): DetectedObject {
+		let bounding_poly: Array<Coordinate> = object
+			.boundingPoly!.normalizedVertices!.map(VisionExtractionProvider.createBoundingVertex)
+			.filter((coord) => {
+				if (coord.x == 0 && coord.y == 0) {
```

Comment:
Can the bounding polygon ever have a vertex on `(0, 0)`? If it can, a valid vertex would be removed by this check.